core/internal/example/example.proto

```diff
@@ -0,0 +1,56 @@
+syntax = "proto3";
+
+// package example provides a very simple example module for registering
+// a name with an owner.
+package example;
+
+import "cosmos/app/v1alpha1/module.proto";
+import "cosmos/msg/v1/msg.proto";
+
+message Module {
+  option (cosmos.app.v1alpha1.module) = {
+    go_import: "github.com/cosmos/cosmos-sdk/core/internal/example"
+    use_package: {
+      name: "example"
+    }
+  };
+}
+
+service Msg {
+  option (cosmos.msg.v1.service) = true;
+
+  // RegisterName registers some arbitrary name with the signer of the
+  // transaction as the name's owner.
+  rpc RegisterName(MsgRegisterName) returns (MsgRegisterNameResponse);
+}
+
+message MsgRegisterName {
+  option (cosmos.msg.v1.signer) = "sender";
+
+  string sender = 1;
```

Comment:
maybe out of scope for this PR, but i feel like i'd prefer to do `string sender = 1 [cosmos.msg.v1.signer = true];`. could be annoying dealing with the occasional misspelling of the field you want to mark as signer 🤷🏻

Reply:
seems like maybe that ship has already sailed...